src/commands/ban.ts

```diff
@@ -12,11 +12,20 @@ async function banHandler(interaction: ChatInputCommandInteraction): Promise<voi
 
 	const guild = await interaction.guild!.fetch();
 	const executor = interaction.user;
-	const users = interaction.options.getString('users', true);
+	const subcommand = interaction.options.getSubcommand();
 	const reason = interaction.options.getString('reason', true);
 	const deleteMessages = interaction.options.getNumber('delete_messages');
 
-	const userIds = [...new Set(Array.from(users!.matchAll(new RegExp(MessageMentions.UsersPattern, 'g')), match => match[1]))];
+	let userIds;
+	if (subcommand === 'user') {
+		const user = interaction.options.getUser('user', true);
+		userIds = [user.id];
+	} else if (subcommand === 'multiuser') {
+		const users = interaction.options.getString('users', true);
+		userIds = [...new Set(Array.from(users.matchAll(/\d{17,18}/g), match => match[0]))];
```

Comment:
Maybe I'm missing something but wouldn't it be better to use the regex DiscordJS provides for this?

https://github.com/discordjs/discord.js/blob/5c90b7f7165a9156b87d741a7178eba92368b576/packages/discord.js/src/structures/MessageMentions.js#L49

Reply:
The purpose of this is to match user IDs both inside mentions and outside mentions.

Reply:
Is posting just user IDs something that commonly happens? The intent is to use mentions here, and it's how I've always used/documented the command. So I'm unsure how common this is?

Reply:
Yeah, the mods specifically requested the ability to use user IDs here.

Reply:
Sounds good then :+1: